src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/AddressingStepType.php

```diff
@@ -42,23 +43,23 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 }
             })
             ->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) use ($options) {
-                /* @var $user UserInterface */
-                $user = $options['user'];
-                if (null === $user || !$user instanceof UserInterface) {
+                /* @var CustomerInterface $customer */
+                $customer = $options['customer'];
+                if (null === $customer || !$customer instanceof CustomerInterface) {
                     return;
                 }
 
                 /* @var $order OrderInterface */
                 $order = $event->getData();
-                if ($order->getShippingAddress() === null && $user->getShippingAddress() !== null) {
-                    $address = clone $user->getShippingAddress();
-                    $address->setUser(null);
+                if ($order->getShippingAddress() === null && $customer->getShippingAddress() !== null) {
```

Comment:
This might be where you need to be careful about protecting a non-registered customer's address details. 

Is there an isUser() method on a customer?